go/worker/client/worker.go

```diff
@@ -131,14 +137,19 @@ func (w *Worker) registerRuntime(commonNode *committeeCommon.Node) error {
 
 	switch config.GlobalConfig.Mode {
 	case config.ModeClient, config.ModeStatelessClient:
-		// When a node is a client node and it has an entity configured, we register it with the
-		// observer role as this may be needed for confidential runtimes.
+		// When a node is a client node and it has an entity configured,
+		// we register it with the observer role as this may be needed
+		// for confidential runtimes.
 		rp, err := w.registration.NewRuntimeRoleProvider(node.RoleObserver, id)
 		if err != nil {
 			return fmt.Errorf("failed to create role provider: %w", err)
 		}
 
-		rp.SetAvailable(commonNode.RegisterNodeRuntime)
+		// The role provider is set as unavailable by default and we want to
+		// leave it like that before initialization starts, so the registration
+		// is blocked until the runtime finishes initializing.
+
```

Comment:
```suggestion
```

Reply:
If I remove this blank line, it will look like the assignment below is somehow related to the text in the comment, which it is not.

Reply:
It actually is, because of that reason you are storing this provider locally. You can leave it as it is.